src/pages/vans/Vans.jsx

```diff
@@ -0,0 +1,121 @@
+import { useState } from "react";
+import { Link, useSearchParams, useLoaderData } from "react-router-dom";
+import { getVans } from "../../api";
+
+export function loader() {
+  return getVans();
+}
+
+const Vans = () => {
+  const info = useLoaderData();
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [error, setError] = useState(null);
+  const typeFilter = searchParams.get("type");
+
+  const filtered = typeFilter
+    ? info.filter((char) => char.type === typeFilter)
+    : info;
+
+  const handleFilterChange = (key, value) => {
+    setSearchParams((prevParams) => {
+      if (value === null) {
+        prevParams.delete(key);
+      } else {
+        prevParams.set(key, value);
+      }
+      return prevParams;
+    });
+  };
+
+  if (error) {
+    return (
+      <h1 className="text-2xl font-black text-[#161616] bg-[#FFF7ED] leading-6">
+        There was an error: {error.message}
+      </h1>
+    );
+  }
+
+  return (
+    <>
+      <div className="bg-[#fff7ed] pb-14">
+        <div className="max-w-7xl mx-auto px-4">
+          <h1 className="font-bold text-3xl leading-8 text-[#161616] pt-14 mb-6">
+            Explore our van options
+          </h1>
+          <div className="flex justify-between">
+            {[
+              { name: "Simple", type: "simple" },
+              { name: "Luxury", type: "luxury" },
+              { name: "Rugged", type: "rugged" },
+            ].map((item, index) => {
+              return (
+                <div
+                  className={` cursor-pointer px-3 py-2 rounded-md text-[#4D4D4D]  text-center text-base leading-6
+                  ${
+                    typeFilter === item.type
+                      ? item.type === "luxury"
+                        ? "text-white bg-[#161616]"
+                        : item.type === "simple"
+                        ? "text-white bg-[#e17654]"
+                        : "text-white bg-[#115e59]"
+                      : "bg-[#FFEAD0]"
+                  }
+                   `}
+                  onClick={() => handleFilterChange("type", item.type)}
+                  key={index}
+                >
+                  {item.name}
+                </div>
+              );
+            })}
+            {typeFilter ? (
+              <p
+                onClick={() => handleFilterChange("type", null)}
+                className="text-[#4D4D4D] font-medium cursor-pointer text-base leading-6 underline"
+              >
+                Clear filter
+              </p>
+            ) : null}
+          </div>
+          <div className="grid grid-cols-2 gap-8 mt-14">
+            {filtered.map((item) => {
+              return (
```

Comment:
I can suggest avoiding using too many nested returns, which can  end up being hard to read, you can use brackets to enclose the JSX element,
```suggestion
            {filtered.map((item) => (
```